ARCs/arc-0056.md

```diff
@@ -0,0 +1,376 @@
+
+## Specification
+
+### Contract Interface
+
+Every application is described via the following interface which is an extension of the `Contract` interface described in [ARC-4](./arc-0004.md).
+
+```ts
+/** Describes the entire contract. This interface is an extension of the interface described in ARC-4 */
+interface Contract {
+  /** The ARCs used and/or supported by this contract. All contracts implicity support ARC4 and ARC56 */
+  arcs: number[];
+  /** A user-friendly name for the contract */
+  name: string;
+  /** Optional, user-friendly description for the interface */
+  desc?: string;
+  /**
+   * Optional object listing the contract instances across different networks
+   */
+  networks?: {
+    /**
+     * The key is the base64 genesis hash of the network, and the value contains
+     * information about the deployed contract in the network indicated by the
+     * key. A key containing the human-readable name of the network MAY be
+     * included, but the corresponding genesis hash key MUST also be defined
+     */
+    [network: string]: {
+      /** The app ID of the deployed contract in this network */
+      appID: number;
+    };
+  };
+  /** Named structs use by the application */
+  structs: { [structName: StructName]: StructFields };
+  /** All of the methods that the contract implements */
+  methods: Method[];
+  state: {
+    /** Defines the values that should be used for GlobalNumUint, GlobalNumByteSlice, LocalNumUint, and LocalNumByteSlice when creating the application  */
+    schema: {
+      global: {
+        ints: number;
+        bytes: number;
+      };
+      local: {
+        ints: number;
+        bytes: number;
+      };
+    };
+    /** Mapping of human-readable names to StorageKey objects */
+    keys: {
+      global: { [name: string]: StorageKey };
+      local: { [name: string]: StorageKey };
+      box: { [name: string]: StorageKey };
+    };
+    /** Mapping of human-readable names to StorageMap objects */
+    maps: {
+      global: { [name: string]: StorageMap };
+      local: { [name: string]: StorageMap };
+      box: { [name: string]: StorageMap };
+    };
+  };
+  /** Supported bare actions for the contract. An action is a combination of call/create and an OnComplete */
+  bareActions: {
+    /** OnCompletes this method allows when appID === 0 */
+    create: ("NoOp" | "OptIn" | "DeleteApplication")[];
+    /** OnCompletes this method allows when appID !== 0 */
+    call: (
+      | "NoOp"
+      | "OptIn"
+      | "CloseOut"
+      | "ClearState"
+      | "UpdateApplication"
+      | "DeleteApplication"
+    )[];
+  };
+  /** Information about the TEAL */
+  sourceInfo?: SourceInfo[];
```

Comment:
Should this be keyed by approval vs clear?

Reply:
Yeah good call. Will update.